### PR TITLE
New Script: Grist

### PR DESCRIPTION
**Pull request by cfurrow**
> **🛠️ Note:**  
> We are meticulous about merging code into the main branch, so please understand that pull requests not meeting the project's standards may be rejected. It's never personal!  
> 🎮 **Note for game-related scripts:** These have a lower likelihood of being merged.

---

## ✍️ Description
Adding new script to install and update Grist (https://www.getgrist.com/). 

Note: I'm new to Proxmox and Helper-Scripts, and I've not used Grist until I began to create these scripts. I wanted to try out Grist, and I had just recently created a Proxmox node, so I thought I'd try my hand at creating helper scripts to get Grist up and running on my Proxmox node.

I used the homarr scripts/json as a guide to create this PR.

---

## 🛠️ Type of Change
Please check the relevant options:  
- [ ] Bug fix (non-breaking change that resolves an issue)  
- [ ] New feature (non-breaking change that adds functionality)  
- [ ] Breaking change (fix or feature that would cause existing functionality to change unexpectedly)  
- [x] New script (a fully functional and thoroughly tested script or set of scripts)  

---

## ✅ Prerequisites
The following steps must be completed for the pull request to be considered:  
- [x] Self-review performed (I have reviewed my code to ensure it follows established patterns and conventions.)  
- [x] Testing performed (I have thoroughly tested my changes and verified expected functionality.)
   - In another branch I did a find/replace to update wget/source calls from `community-scripts/ProxmoxVE/raw/main` to my test branch, `cfurrow/ProxmoxVE/tree/grist-dev`. Then I was able to verify the initial install, and update scripts worked.
- [ ] Documentation updated (I have updated any relevant documentation)
    - ⚠️ I don't know if I've done this, and I don't know what other documentation outside of the new `grist.json` would need updating? Maybe new scripts do not need doc updates?

---

## 📋 Additional Information (optional)

As mentioned in the above description--I'm new to Proxmox, and Helper Scripts, so I'd appreciate any guidance and feedback on this PR. I'm hoping most of it is OK since I copied from the existing "homarr" scripts.

Something that I hope to see in the future for the helper scripts is a tutorial/guide for writing new scripts, as well as some additional quality of life updates for developers. Having to manually change the URLs so that your local proxmox will use your own repo files is a bit cumbersome, as you then have to revert those changes before pushing up to your final branch. Maybe there is already a way to "bootstrap" some basic details, such as the base repo which could obviate the need to do a find/replace and instead set those details in a `.env` file or similar. Simply spitballing and this may already be underway by the community.

### CODE REVIEW

File: ct/grist.sh [hunk 1]
```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+source <(curl -s https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/misc/build.func)
+# TODO: create frontend json files, etc for Grist
```
Comment (cfurrow) [inline ct/grist.sh:3]: Remove. This has been done.

File: install/grist-install.sh [hunk 2]
```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+
+# Copyright (c) 2021-2024 tteck
+# Author: tteck (tteckster)
```
Comment (cfurrow) [inline install/grist-install.sh:4]: update to cfurrow ?

File: install/grist-install.sh [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+update_os
+
+msg_info "Installing Dependencies"
+$STD apt-get install -y \
+  curl \
+  sudo \
+  make \
+  gnupg \
+  ca-certificates \
+  mc \
```
Comment (cfurrow) [inline install/grist-install.sh:24]: I don't know if this is required. `homarr` (and other scripts) seem to add it as a dep, but I don't know how/if it's used. Maybe it's more of a courtesy include for the community?

File: install/grist-install.sh [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+msg_info "Installing Grist"
+RELEASE=$(curl -s https://api.github.com/repos/gristlabs/grist-core/releases/latest | grep "tag_name" | awk '{print substr($2, 3, length($2)-4) }')
+cd /opt
+wget -q https://github.com/gristlabs/grist-core/archive/refs/tags/v${RELEASE}.zip
+unzip -q v$RELEASE.zip
+rm -rf v${RELEASE}.zip
+mv grist-core-${RELEASE} grist
+cd grist
+$STD yarn install
+$STD yarn run build:prod
```
Comment (cfurrow) [inline install/grist-install.sh:52]: The install instructions from https://github.com/gristlabs/grist-core include an additional command, `yarn run install:python` but it is not necessary given I install `python3.11-venv`, which install python 3.11

File: install/grist-install.sh [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+echo "${RELEASE}" >/opt/${APPLICATION}_version.txt
+
+cat <<EOF >/opt/grist/.env
+NODE_ENV=production
+GRIST_HOST=0.0.0.0
```
Comment (cfurrow) [inline install/grist-install.sh:57]: Required, or else Grist binds to `localhost`, which makes the grist server unavailable when trying to browse to the CT's IP and port (e.g. http://192.168.10.115:8484)

File: json/grist.json [hunk 6]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "Grist",
+  "slug": "grist",
+  "categories": [
+    5
```
Comment (cfurrow) [inline json/grist.json:5]: Databases category.

File: install/grist-install.sh [hunk 7]
```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+
+# Copyright (c) 2021-2024 tteck
```
Comment (michelroegl-brunner) [inline install/grist-install.sh:3]:
```suggestion

#Copyright (c) 2021-2024 community-scripts ORG

```

File: install/grist-install.sh [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+msg_info "Setting up Node.js Repository"
+mkdir -p /etc/apt/keyrings
+curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg
+echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_20.x nodistro main" >/etc/apt/sources.list.d/nodesource.list
+msg_ok "Set up Node.js Repository"
+
+msg_info "Installing Node.js"
+$STD apt-get update
+$STD apt-get install -y nodejs
+msg_ok "Installed Node.js"
```
Comment (michelroegl-brunner) [inline install/grist-install.sh:37]: 28-37, put in one Section "Installing Node.js"

File: install/grist-install.sh [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+msg_info "Installing yarn"
+$STD npm install -g yarn
+msg_ok "Installed yarn $(yarn --version)"
+
+msg_info "Installing Grist"
+RELEASE=$(curl -s https://api.github.com/repos/gristlabs/grist-core/releases/latest | grep "tag_name" | awk '{print substr($2, 3, length($2)-4) }')
+cd /opt
+wget -q https://github.com/gristlabs/grist-core/archive/refs/tags/v${RELEASE}.zip
+unzip -q v$RELEASE.zip
+rm -rf v${RELEASE}.zip
```
Comment (michelroegl-brunner) [inline install/grist-install.sh:48]: Put under cleanup

**Review by michelroegl-brunner — CHANGES_REQUESTED** [review]
Little review under mobile, I have another look tommorow.

File: install/grist-install.sh (hunk 3 above)
Comment (michelroegl-brunner) [reply install/grist-install.sh:24]: Yes exactly. tteck started it and we keep it.

**Comment by cfurrow** [comment]
Amazing, @michelroegl-brunner! I'll make those changes, but there is absolutely no rush on this.

File: install/grist-install.sh (hunk 8 above)
Comment (cfurrow) [reply install/grist-install.sh:35]: Done -- good call

**Comment by cfurrow** [comment]
Moving back to draft PR. I noticed the `update` script was not working as expected. Previous manual tests were working, but something has broken since then. Investigating.

**Review by MickLesk — APPROVED** [review]

**Review by michelroegl-brunner — APPROVED** [review]

**Comment by MickLesk** [comment]
@dsiebel @michelroegl-brunner 

the script has some issues. Ive fixed asap some things. But the ressources are to low. 
 Memory usage 97.89% (1002.37 MiB of 1.00 GiB)
 => Break with heap space of nodejs

Increase the RAM to 2GB for Website & CT
=> Create an new PR based on main changes, because i already had do some things

**Comment by michelroegl-brunner** [comment]
Changed things arround, installs and runs. But i do not see any missing indentations? Also VS Code Formate dose not change anything.

**Pull request merged**